app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt

```diff
@@ -158,13 +162,19 @@ class EventsFragment : Fragment() {
         super.onViewCreated(view, savedInstanceState)
 
         val eventClickListener: EventClickListener = object : EventClickListener {
-            override fun onClick(eventID: Long) { EventDetailsFragmentArgs.Builder()
-                .setEventId(eventID)
-                .build()
-                .toBundle()
-                .also { bundle ->
-                    findNavController(view).navigate(R.id.eventDetailsFragment, bundle, getAnimFade())
+            override fun onClick(eventID: Long, sharedImage: ImageView) {
```

Comment:
@liveHarshit i think I'm using the navigation architecture

Reply:
https://developer.android.com/guide/navigation/navigation-animate-transitions#kotlin
https://stackoverflow.com/questions/50599360/how-to-use-shared-element-transitions-in-navigation-controller

Reply:
@liveHarshit the above method won't work with recyclerview as for recyclerview transitionName can't be set to xml and i've the somewhat similiar method

Reply:
@liveHarshit check this out https://stackoverflow.com/questions/53614436/how-to-implement-shared-transition-element-form-rcyclerviews-item-to-fragment-w